public/app/features/query/state/PanelQueryRunner.ts

```diff
@@ -338,11 +338,12 @@ export class PanelQueryRunner {
 }
 
 async function getDataSource(
-  datasource: DataSourceRef | string | DataSourceApi | null,
+  datasource: DataSourceRef | DataSourceApi | null,
```

Comment:
I (inadvertently) changed the signature here to remove `string` from the type, but the `datasource` variable coming in is only typed to `datasource: DataSourceRef | DataSourceApi<TQuery, TOptions> | null;` so I think it's fine?

Reply:
Unfortunately there's some dodgy typing going on higher up. This function actually is called with a string at runtime, which is not apparent at all 😓